### ISSUE TITLE
FinalLocalVariable: "return" false negative

### ISSUE THREAD

**Issue by Luolc** [checkstyle/checkstyle#4226]
Taken from #4082 

```
$ javac Foo.java
$ cat Foo.java
public class Foo {
    void foo() {
        int a; // expecte a violation
        final int random = (int) Math.random();
        if (random > 0) {
            a = 1;
            return;
        }
        a = 2;
    }
}
```

```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="FinalLocalVariable">
        </module>
    </module>
</module>
```

```
$ java -jar checkstyle-7.7-SNAPSHOT-all.jar -c config.xml Foo.java
Starting audit...
Audit done.
```

Expect: a violation at line 3.

**Comment by DXTkastb** [comment]
I'm on it.

**Comment by pbludov** [comment]
> I'm on it.

Hold on. The default value for the `format` property is `"^[a-z][a-zA-Z0-9]*$"`.
The name `a` matches this pattern. There is no false negatives.
To issue a violation here, this config should be changed to something like
```xml
        <module name="LocalVariableName">
            <!-- must be at least three characters long -->
            <property name="format" value="[a-z][a-zA-Z0-9]{2,}$"/>
        </module>
```
@Luolc please close this issue.
